app/frontend/actions/reloadWallet.ts

```diff
@@ -24,6 +24,7 @@ export default (store: Store) => {
   const reloadWalletInfo = async (state: State): Promise<void> => {
     setWalletOperationStatusType(state, 'reloading')
     const wallet = getWallet()
+    wallet.invalidateCache()
```

Comment:
Haven't tested, but if I for example click the `reload` button 500x times, it would try to fetch all 500x times right? 
Comparing this to react-query, It seems to me that the invalidation should happen in (after) mutation (tx submission in this case), this is basically like calling the invalidation before querying right?

Reply:
The problem is when the user submits tx, which times out. If we would cache it, then he will spam the reload button and can get into a state, where the transaction is submitted, but not fetched, because it will use the cached value.

Regarding fetching it 500 times, this function is already debounced.

Reply:
I see that the debouncing fixes the issue with clicking the reload button multiple times. I am fine with the current state, but conceptually its still seems to me that we should call the invalidation in the `waitForTxToAppearOnBlockchain` just before the reload and in the `catch` of the `submitTx` fn (or some other way which would fix the cache issue you mentioned, I guess there is a way to do it). Feel free to leave it as it is. :+1:

Reply:
Currently we rely on the whole account info being reloaded by `reloadWalletInfo` therefore I'd lean towards keeping the invalidation there (to ensure everything is fetched is sync) rather than ad-hoc invalidating the cache (which is a valid approach, but given how simplistic the caching is, I can see that getting out of hand)

Reply:
basically, the purpose of caching here is to just prevent re-fetching data by calls depending on the same data while we (re)load the wallet and in that context it IMHO makes sense to manage the cache from there

Reply:
Makes sense, thnx :+1: